src/build.d

```diff
@@ -679,15 +679,17 @@ void processEnvironment()
 {
     const os = env["OS"];
 
-    auto hostDMDVersion = [env["HOST_DMD_RUN"], "--version"].execute.output;
-    if (hostDMDVersion.canFind("DMD"))
+    const hostDMDVersion = [env["HOST_DMD_RUN"], "--version"].execute.output;
+    const kindIdx = hostDMDVersion.canFind("DMD", "LDC", "GDC", "gdmd", "gdc");
+
+    enforce(kindIdx, "Invalid Host DMD found: " ~ hostDMDVersion);
```

Comment:
Having to keep the `kindIdx` in sync with the argument list above doesn't seem ideal.  What about something like this?
```D
    const hostDMDVersion = [env["HOST_DMD_RUN"], "--version"].execute.output;
    for (string versionRemaining = hostDMDVersion; ; versionRemaining.popFront) {
        enforce(!versionRemaining.empty, "Invalid Host DMD found: " ~ hostDMDVersion);
        if (versionRemaining.startsWith("DMD")) {
            env["HOST_DMD_KIND"] = "dmd";
            break;
        } else if (versionRemaining.startsWith("LDC")) {
            env["HOST_DMD_KIND"] = "ldc";
            break;
        } else if (versionRemaining.startsWith("GDC") ||
            versionRemaining.startsWith("gmd") ||
            versionRemaining.startsWith("gdb")) {
            env["HOST_DMD_KIND"] = "gdc";
            break;
        }
    }
```

Reply:
I removed the magic numbers so we can add/remove other identifiers (e.g. "dmd") without breaking the evaluation of kindIdx.
If you would still prefer your proposed solution i can use that approach as well.

Reply:
Clever.  I like it better than the "magic numbers" solution.

I think the solution I proposed is a bit less brittle as you don't have to keep the "ordering" in sync and if you add a new case, you don't have to modify the other cases, i.e. with your solution if you added another kind before `DMD` or `LDC` then you would need to modify the `if` conditions for `DMD` and `LDC`. I'll defer to @wilzbach on the final decision.

Reply:
True. Im fine with either solution as long as it works

Reply:
@marler8997's solution is more elegant, but this is a build script that is supposed to be read by humans and there are only three compiler. This won't change in the foreseeable future, so I think we are fine with @MoonlightSentinel's solution. Thanks!

Reply:
Thank you @wilzbach and @marler8997.